### PR TITLE
Sio2bt

### PR DESCRIPTION
**Pull request by TheMontezuma**
As discussed, I rebased SIO2BT modifications on top of your lastest commit.
Please start review with Makefile and README.txt modifications.

For simplicity I duplicated the code:
hisiodet.src -> hisiodetbt.src
where major differences are located (otherwise BT flag is used to tweak the code)

There will be two seperate binaries generated:
hipatch-code-rom.bin
hipatch-code-rom-bt.bin

No patching for SIO2BT at runtime (hisio.com, etc.) is possible.

### CODE REVIEW

File: hisiodetbt.src [hunk 1]
```diff
@@ -0,0 +1,182 @@
+;  highsiodet.src - check for highspeed drives
```
Comment (HiassofT) [inline hisiodetbt.src]: can you change that to match the filename and add a short note that it applies only to BT?

File: hisiodetbt.src [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+;
+;  Copyright (c) by Matthias Reichl <hias@horus.com>
```
Comment (HiassofT) [inline hisiodetbt.src:3]: Please add a copyright entry with your name and date (eg 2019 or 2019-2020)

**Review by HiassofT — CHANGES_REQUESTED** [review]

**Comment by HiassofT** [comment]
Thanks, in general the PR is fine, I added a few comments.

Can you please also change the author in the commits to your real name?

I also noticed I had unpushed changes, adding the direct highspeed SIO entry point, can you add the label? Just rebase and squash this commit https://github.com/HiassofT/highspeed-sio/commit/19e20690ccc764244803dc374f41c23eeee6385e in

I've been thinking about adding the "SIO2BT" string into the ROM instead of the version date I have, if the patch is compiled with BT - then we have a marker and it's easy to check which patch is in the ROM. I can handle that separately though. What do you think?

File: hisiodetbt.src (hunk 1 above)
Comment (TheMontezuma) [reply hisiodetbt.src]: Sure

File: hisiodetbt.src (hunk 2 above)
Comment (TheMontezuma) [reply hisiodetbt.src:3]: OK

**Comment by TheMontezuma** [comment]
I hope this is what you have asked me for :)

**Comment by HiassofT** [comment]
can you rebase on top of master and force-push the PR branch? That should get rid of bt merge commit and I think then everything should be fine

**Comment by HiassofT** [comment]
ah, better use git rebase -i and squash/fixup the "correct author and filename" commit into the "add SIO2BT" commit.

**Comment by HiassofT** [comment]
Just did a local check and noticed that Author in the commits and SOB still contained your nickname:
```
commit 23bdf00d7a733adf1b788704a94e2e7f422e9073 (bt)
Author: TheMontezuma <marcin.sochacki@gmx.net>
Date:   Sat Jul 20 16:38:57 2019 +0200

    add SIO2BT support
    
    Signed-off-by: TheMontezuma <marcin.sochacki@gmx.net>
```
Could you fix that so they contain your real name? git rebase -i and choosing e(dit) on your commits, and then running git commit --amend --author="...." should do the trick.

BTW: I've pushed a squash version of this PR (without any Author/SOB modifications though) to the hisio-bt-test branch

**Comment by TheMontezuma** [comment]
> can you rebase on top of master and force-push the PR branch? That should get rid of bt merge commit and I think then everything should be fine

OK. This time I rebased SIO2BT changes on top of your master branch (and not on top of the latest commit).

**Comment by HiassofT** [comment]
Thanks, the changes look fine now! I've added the DOHIENTRY label back, reinstated my copyright line in the file and pushed the changes to the PR branch.

One last thing: could you test if the direct entry point works with SIO2BT? I think it should work just fine but don't have the HW here to test myself.

A simple test is in the repo https://github.com/HiassofT/highspeed-sio/blob/master/diag-entry.src (build with `make diag-entry.atr`), by default it'll read sectors from a double density ATR in D2: at pokey divisor 8 (these parameters are hardcoded at the beginning of the file).

Just use hisio/bt patch from this branch in the Atari, put diag-entry.atr in D1:, an empty DD ATR in D2: and boot - if everything's fine it'll just load sectors continously

**Comment by TheMontezuma** [comment]
It works as expected :)

IMG_20200512_152800

**Comment by HiassofT** [comment]
awesome, thanks a lot for checking!

**Pull request merged**